src/app/views/login/SignInController.jsx

```diff
@@ -0,0 +1,244 @@
+                    });
+            })
+            .catch(error => {
+                if (error) {
+                    const notification = {
+                        type: "warning",
+                        isDismissable: true,
+                        autoDismiss: 15000
+                    };
+
+                    switch (error.status) {
+                        case 404: {
+                            const email = Object.assign({}, this.state.email, {
+                                errorMsg: "Email address not recognised"
+                            });
+                            this.setState({
+                                email: email,
+                                isSubmitting: false
+                            });
+                            break;
+                        }
+                        case 401: {
+                            const password = Object.assign({}, this.state.email, { errorMsg: "Incorrect password" });
+                            this.setState({
+                                password: password,
+                                isSubmitting: false
+                            });
+                            break;
+                        }
+                        case 417: {
+                            this.setState({
+                                requestPasswordChange: true
+                            });
+                            break;
+                        }
+                        case "UNEXPECTED_ERR": {
+                            console.error(errCodes.UNEXPECTED_ERR);
+                            notification.message = errCodes.UNEXPECTED_ERR;
+                            notifications.add(notification);
+                            break;
+                        }
+                        case "RESPONSE_ERR": {
+                            console.error(errCodes.RESPONSE_ERR);
+                            notification.message = errCodes.RESPONSE_ERR;
+                            notifications.add(notification);
+                            break;
+                        }
+                        case "FETCH_ERR": {
+                            console.error(errCodes.FETCH_ERR);
+                            notification.message = errCodes.FETCH_ERR;
+                            notifications.add(notification);
+                            break;
+                        }
+                    }
+                }
+                this.setState({ isSubmitting: false });
+            });
+    }
+
+    handleSubmit(event) {
+        event.preventDefault();
+
+        const credentials = {
+            email: this.state.email.value,
+            password: this.state.password.value
+        };
+
+        this.setState({ isSubmitting: true });
+
+        this.handleLogin(credentials);
+    }
+
+    handleInputChange(event) {
+        const id = event.target.id;
+        const value = event.target.value;
+
+        switch (id) {
```

Comment:
I know this is a copy/paste of another file but you should always have a `default` case on a switch that throws an error if the other cases don't match—this gives you a bit of a defence against typos, etc

Reply:
100% agree on this and will add to one of the next PRs for the login work.